### PR TITLE
Km modeltweaks

### PR DESCRIPTION
**Pull request by kendalynnm**
This branch:

- finalized data QA/QC
- changed the model structure to include
    - study number as a random effect
    - natural log of response ratio as effect size calculation
    - most important predictors
    - all effect sizes from the three variable type
    - cooks distance test
- graphs for AGU and manuscript
- rearranging and changings in content to RMD to reflect the above changes

### CODE REVIEW

File: MultiYearStudies.R [hunk 1]
```diff
@@ -1,44 +1,53 @@
 
-library(skimr)
 
-#duration
-meta_df %>%
-  group_by(Study_number, OtherFactor, Ecosystem_type) %>%
-  mutate(Duration = Study_midyear - min(Study_midyear)) -> meta_time
+#conduct meta-analysis
+#this is the best fitting model of irrigiation studies - not sure if the "/" notation is correct
```
Comment (bpbond) [inline MultiYearStudies.R:4]: Interesting; I don't know.

File: MultiYearStudies.R [hunk 2]
```diff
@@ -1,44 +1,53 @@
...
 
+res_irrigation <- rma(yi, vi, random = ~ 1|Study_number/Ecosystem_type,
+              mods = ~Duration*Percent_control*Latitude,
+              method = "ML",
+           data = metadat[metadat$Manipulation == "Irrigation",],
+           test = "knha")
+summary.rma(res_irrigation)
+profile(res_irrigation)
+
+
```
Comment (bpbond) [inline MultiYearStudies.R:17]: I comment here would probably be good: what's `rma.mv` doing?

File: MultiYearStudies.R [hunk 3]
```diff
@@ -1,44 +1,53 @@
...
+res_drought <- rma.mv(yi, vi, random = ~ 1|Study_number/Ecosystem_type,
+                         mods = ~Duration * Percent_control * Latitude,
+                         method = "ML",
+                         data = metadat_short[metadat_short$Manipulation == "Drought",])
+summary.rma(res_drought)
+profile(res_drought)
+
+library(MuMIn)
+if (!require("devtools")) {
```
Comment (bpbond) [inline MultiYearStudies.R:26]: Don't do this. (1) The script won't work without `devtools` so there's no point in using `require`. (2) Wait what? We don't want to install `dmetar` every time we run, so line 29 should be _inside_ the if block. (3) It's better to just fail with a message saying where `dmetar` can be found.

File: MultiYearStudies.R [hunk 4]
```diff
@@ -1,44 +1,53 @@
...
-#check the call arguments for rma.mv vs rma
-res <- rma.mv(yi, vi, random = ~1|Study_number,
-           mods= ~Ecosystem_type + Duration + Percent_control - 1,
-           data = metadat[metadat$Manipulation == "Irrigation",])
-res
 
 #conduct meta-analysis
 res2 <- rma(yi, vi, 
-           mods= ~Ecosystem_type + Duration + Percent_control - 1,
+           mods= ~Ecosystem_type*Duration*Percent_control - 1,
```
Comment (bpbond) [inline MultiYearStudies.R:46]: Put spaces around asterisks please, for readability.

File: Scratch.R [hunk 5]
```diff
@@ -0,0 +1,121 @@
+
```
Comment (bpbond) [inline Scratch.R:1]: Is this just a scratchpad? Should it be reviewed?

File: getSoilClay.R [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+library(sf)
+
+url = "https://files.isric.org/soilgrids/latest/data/"
+
+# variable of interest
+voi = "clay"
+depth = "15-30cm"
+quantile = "mean"
+
+variable = paste(url, voi, sep="")
```
Comment (bpbond) [inline getSoilClay.R:15]: FYI there's a `paste0` function that's simpler to use.

**Review by bpbond — CHANGES_REQUESTED** [review]
Looks good @kendalynnm ! A few minor things, mostly around clarity.

File: MultiYearStudies.R (hunk 3 above)
Comment (stephpenn1) [reply MultiYearStudies.R:26]: +1 to Ben's comments, typically we only use `install.packages()` on the console line or in one of these if statements. For your purposes i would just remove the if statement and if you want to make sure users load `devtools` and `dmetar`, add it as a `library()` call

**Review by stephpenn1 — CHANGES_REQUESTED** [review]

File: MultiYearStudies.R (hunk 1 above)
Comment (kendalynnm) [reply MultiYearStudies.R]: This is a legacy of my sloppy coding, apologies! I've updated the comments in this file to accurately reflect what's on the script and it's role in the project.

File: MultiYearStudies.R (hunk 3 above)
Comment (kendalynnm) [reply MultiYearStudies.R]: Also here, this is a legacy of my sloppy coding, apologies! This was a dead end that was ready to be deleted (and now has been).

File: MultiYearStudies.R (hunk 2 above)
Comment (kendalynnm) [reply MultiYearStudies.R]: Done, but in the main file (rs-synthesis.Rmd).

File: MultiYearStudies.R (hunk 4 above)
Comment (kendalynnm) [reply MultiYearStudies.R]: Done.

File: Scratch.R (hunk 5 above)
Comment (kendalynnm) [reply Scratch.R:1]: No need to review this, thanks!

File: getSoilClay.R (hunk 6 above)
Comment (kendalynnm) [reply getSoilClay.R]: file no longer exsists

File: MultiYearStudies.R [hunk 7]
```diff
@@ -1,44 +1,53 @@
...
+
+
+res_drought <- rma.mv(yi, vi, random = ~ 1|Study_number/Ecosystem_type,
+                         mods = ~Duration * Percent_control * Latitude,
+                         method = "ML",
+                         data = metadat_short[metadat_short$Manipulation == "Drought",])
+summary.rma(res_drought)
+profile(res_drought)
+
+library(MuMIn)
```
[the remark this answers is not in the document]
Comment (kendalynnm) [reply MultiYearStudies.R]: This is a legacy of my sloppy coding, apologies! This part of the script has been deleted.

**Comment by bpbond** [comment]
Woot woot @kendalynnm ! 👏

**Pull request merged**